script-test/playbackstrategies/msestrategytest.js

```diff
@@ -1,491 +1,579 @@
 
-          expect(mseStrategy.getDuration()).toBe(101);
-        });
+            mseStrategy.load('src', null, 60);
 
-        it('returns 0 when MediaPlayer is undefined', function () {
-          setUpMSE();
+            expect(mockDashInstance.initialize).toHaveBeenCalledWith(mockVideoElement, 'src#t=160', true);
+          });
 
-          expect(mseStrategy.getDuration()).toBe(0);
-        });
-      });
+          it('should playback from the live point of a webcast', function () {
+            setUpMSE(0, WindowTypes.GROWING, MediaKinds.VIDEO, 100000, 200000);
 
-      describe('tearDown()', function () {
-        it('should reset the MediaPlayer', function () {
-          setUpMSE();
-          mseStrategy.load('src', null, 0);
+            mockDashInstance.getSource.and.returnValue('src');
 
-          mseStrategy.tearDown();
+            mseStrategy.load('src', null, undefined);
 
-          expect(mockDashInstance.reset).toHaveBeenCalled();
-        });
+            expect(mockDashInstance.initialize).toHaveBeenCalledWith(mockVideoElement, 'src', true);
+          });
 
-        it('should tear down bindings to MediaPlayer Events correctly', function () {
-          setUpMSE();
-          mseStrategy.load(null, null, 0);
-
-          mseStrategy.tearDown();
-
-          expect(mockVideoElement.removeEventListener).toHaveBeenCalledWith('timeupdate', jasmine.any(Function));
-          expect(mockVideoElement.removeEventListener).toHaveBeenCalledWith('playing', jasmine.any(Function));
-          expect(mockVideoElement.removeEventListener).toHaveBeenCalledWith('pause', jasmine.any(Function));
-          expect(mockVideoElement.removeEventListener).toHaveBeenCalledWith('waiting', jasmine.any(Function));
-          expect(mockVideoElement.removeEventListener).toHaveBeenCalledWith('seeking', jasmine.any(Function));
-          expect(mockVideoElement.removeEventListener).toHaveBeenCalledWith('seeked', jasmine.any(Function));
-          expect(mockVideoElement.removeEventListener).toHaveBeenCalledWith('ended', jasmine.any(Function));
-          expect(mockVideoElement.removeEventListener).toHaveBeenCalledWith('error', jasmine.any(Function));
-          expect(mockDashInstance.off).toHaveBeenCalledWith(dashjsMediaPlayerEvents.ERROR, jasmine.any(Function));
-          expect(mockDashInstance.off).toHaveBeenCalledWith(dashjsMediaPlayerEvents.QUALITY_CHANGE_RENDERED, jasmine.any(Function));
-          expect(mockDashInstance.off).toHaveBeenCalledWith(dashjsMediaPlayerEvents.METRIC_ADDED, jasmine.any(Function));
-          expect(mockDashInstance.off).toHaveBeenCalledWith(dashjsMediaPlayerEvents.MANIFEST_LOADED, jasmine.any(Function));
-          expect(mockDashInstance.off).toHaveBeenCalledWith(dashjsMediaPlayerEvents.MANIFEST_VALIDITY_CHANGED, jasmine.any(Function));
-        });
+          it('returns the correct duration from the DASH Mediaplayer with audio', function () {
+            setUpMSE(null, null, MediaKinds.AUDIO);
```

Comment:
I don't think this test is necessary, as we retrieve the duration from the same dash.js API regardless of whether it's audio or video.

I'm assuming this test was created because the dash.js docs were unclear on duration for audio? I can see why that might prompt you to write a test for it, and it would make sense in an integration test that used dash.js itself, but doesn't make a difference when we've created a mock dash.js that has the same behaviour for `duration()` for both audio and video.